FlagsmithClient/Classes/Flagsmith.swift

```diff
@@ -12,269 +12,291 @@ import FoundationNetworking
-          completion(.success(flags))
-        case .failure(let error):
-          if self.defaultFlags.isEmpty {
-            completion(.failure(error))
-          }
-          else {
-            completion(.success(self.defaultFlags))
-          }
+        set {
+            apiManager.propertiesSerialAccessQueue.sync {
+                _defaultFlags = newValue
+            }
         }
-      }
     }
-  }
-  
-  /// Check feature exists and is enabled optionally for a specific identity
-  ///
-  /// - Parameters:
-  ///   - id: ID of the feature
-  ///   - identity: ID of the user (optional)
-  ///   - completion: Closure with Result which contains Bool in case of success or Error in case of failure
-  public func hasFeatureFlag(withID id: String,
-                             forIdentity identity: String? = nil,
-                             completion: @escaping (Result<Bool, Error>) -> Void) {
-    analytics.trackEvent(flagName: id)
-    getFeatureFlags(forIdentity: identity) { (result) in
-      switch result {
-      case .success(let flags):
-        let hasFlag = flags.contains(where: {$0.feature.name == id && $0.enabled})
-        completion(.success(hasFlag))
-      case .failure(let error):
-        if self.defaultFlags.contains(where: {$0.feature.name == id && $0.enabled}) {
-          completion(.success(true))
+
+    /// Configuration class for the cache settings
+    private var _cacheConfig: CacheConfig = CacheConfig()
+    public var cacheConfig: CacheConfig  {
+        get {
+            apiManager.propertiesSerialAccessQueue.sync { _cacheConfig }
         }
-        else {
-          completion(.failure(error))
+        set {
+            apiManager.propertiesSerialAccessQueue.sync {
+                _cacheConfig = newValue
+            }
         }
-      }
     }
-  }
-  
-  /// Get remote config value optionally for a specific identity
-  ///
-  /// - Parameters:
-  ///   - id: ID of the feature
-  ///   - identity: ID of the user (optional)
-  ///   - completion: Closure with Result which String in case of success or Error in case of failure
-  @available(*, deprecated, renamed: "getValueForFeature(withID:forIdentity:completion:)")
-  public func getFeatureValue(withID id: String,
-                              forIdentity identity: String? = nil,
-                              completion: @escaping (Result<String?, Error>) -> Void) {
-    analytics.trackEvent(flagName: id)
-    getFeatureFlags(forIdentity: identity) { (result) in
-      switch result {
-      case .success(let flags):
-        let flag = flags.first(where: {$0.feature.name == id})
-        completion(.success(flag?.value.stringValue))
-      case .failure(let error):
-        if let flag = self.getFlagUsingDefaults(withID: id, forIdentity: identity) {
-          completion(.success(flag.value.stringValue))
+
+    private init() {
+        apiManager = APIManager()
+        analytics = FlagsmithAnalytics(apiManager: apiManager)
+    }
```

Comment:
It allows the analytics manager to be a let rather than a lazy var (guaranteed non-null instead of nullable). This was probably added to make the class work better with Sendable - A type whose values can safely be passed across concurrency domains by copying.